tests/run-make/static-pie/rmake.rs

```diff
@@ -0,0 +1,48 @@
+// How to manually run this
+// $ ./x.py test --target x86_64-unknown-linux-[musl,gnu] tests/run-make/static-pie
+
+//@ only-x86_64
+//@ only-linux
+//@ ignore-32bit
+
+use std::process::Command;
+
+use run_make_support::llvm_readobj;
+use run_make_support::rustc;
+use run_make_support::{cmd, env_var};
+
+fn ok_compiler_version(compiler: &str) -> bool {
+    let check_file = format!("check_{compiler}_version.sh");
```

Comment:
Discussion: I really don't like random shell scripts, but I'm not going to block this PR on porting the shell scripts initially. If you don't want to port the shell scripts, please leave a `// FIXME(jieyouxu): port the check_compiler_version.sh scripts` comment.